sites/public/src/components/browse/FilterDrawerHelpers.tsx

```diff
@@ -0,0 +1,351 @@
+      if (userSelections["minRent"]) {
+        const filter = {
+          $comparison: EnumListingFilterParamsComparison[">="],
+        }
+        filter[ListingFilterKeys.monthlyRent] = userSelections["minRent"]?.replace(",", "")
+        filters.push(filter)
+      }
+      if (userSelections["maxRent"]) {
+        const filter = {
+          $comparison: EnumListingFilterParamsComparison["<="],
+        }
+
+        filter[ListingFilterKeys.monthlyRent] = userSelections["maxRent"]?.replace(",", "")
+        filters.push(filter)
+      }
+    }
+  })
+  return filters
+}
+
+export const isFiltered = (contextQuery: ParsedUrlQuery) => {
+  return Object.keys(contextQuery).some((param) => Object.keys(ListingFilterKeys).includes(param))
+}
+
+export const getFilterQueryFromURL = (url: ParsedUrlQuery) => {
+  delete url["page"]
+  return encode(url)
+}
+
+export const encodeFilterDataToQuery = (data: FilterData): string => {
+  const queryArr = []
+  const cleanedFilterData = removeUnselectedFilterData(data)
+  Object.entries(cleanedFilterData).forEach(([filterType, userSelections]) => {
+    if (arrayFilters.includes(ListingFilterKeys[filterType])) {
+      const arrParam = `${ListingFilterKeys[filterType]}=${Object.keys(userSelections).join(",")}`
+      queryArr.push(arrParam)
+    } else if (booleanFilters.includes(ListingFilterKeys[filterType])) {
+      const booleanParam = `${ListingFilterKeys[filterType]}=true`
+      queryArr.push(booleanParam)
+    } else if (
+      (filterType === ListingFilterKeys.monthlyRent && userSelections["minRent"]) ||
+      userSelections["maxRent"]
+    ) {
+      const rentParam = `${ListingFilterKeys[filterType]}=${Object.values(userSelections).join(
+        "-"
+      )}`
+      queryArr.push(rentParam)
+    }
+  })
+  return queryArr.join("&")
+}
+
+export const decodeQueryToFilterData = (parsedQuery: ParsedUrlQuery): FilterData => {
+  const filterData = {}
+  Object.entries(parsedQuery).forEach(([filterType, userSelections]) => {
+    if (arrayFilters.includes(ListingFilterKeys[filterType])) {
+      typeof userSelections === "string" &&
+        userSelections.split(",").forEach((userSelection) => {
+          if (filterData[filterType]) {
+            filterData[filterType][userSelection] = true
+          } else {
+            filterData[filterType] = { [userSelection]: true }
+          }
+        })
+    } else if (booleanFilters.includes(ListingFilterKeys[filterType]) && isTrue(userSelections)) {
+      filterData[filterType] = true
+    } else if (filterType === ListingFilterKeys.monthlyRent && typeof userSelections === "string") {
+      //custom separator to avoid conflicts with higher values with commas
+      const rentArr = userSelections.split("-")
+      filterData[filterType] = { minRent: rentArr[0], maxRent: rentArr[1] }
+    }
+  })
+  return filterData
+}
+
+export const removeUnselectedFilterData = (data: FilterData): FilterData => {
```

Comment:
nit: some of these functions aren't the most intuitive on what they are doing. Can you add a function JSDoc for this and the other function in this file. Also lets make sure tests will be added for this so can you add `it.todo`s for this function in `FilterDrawerHelpers.test.tsx`